packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -718,10 +695,18 @@ export class LocalFluidDataStoreContextBase extends FluidDataStoreContext {
     }
 
     public generateAttachMessage(): IAttachMessage {
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const entries = this.channel!.getAttachSnapshot();
+        assert(this.channel !== undefined, "There should be a channel when generating attach message");
 
-        const snapshot: ITree = { entries, id: null };
+        let snapshot: ITree;
+        if (this.channel.getAttachSummary !== undefined) {
```

Comment:
please add // back-compat on top of it with a bit more info, such that we can find these cases and remove in future